From 121ba463a32601b1f0a8ea7d053f026ab0399cbf Mon Sep 17 00:00:00 2001
From: Mario Limonciello <mario.limonciello@amd.com>
Date: Fri, 24 Sep 2021 12:32:06 -0500
Subject: [PATCH 38/58] SUNLIGHT: ACPI: PM: s2idle: Don't report missing
 devices as failing constraints

Description:
 - ACPI tables may have entries for devices that are not physically
   present but that can be connected.  These devices shouldn't cause
   constraints checking to fail.

Bug: N/A
Change-Id: I34f5ca978aab69ff0a0906191eec21649b19fe27
Signed-off-by: Ionut Nechita <ionut_n2001@yahoo.com>
Signed-off-by: Mario Limonciello <mario.limonciello@amd.com>
---
 drivers/acpi/x86/s2idle.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/acpi/x86/s2idle.c b/drivers/acpi/x86/s2idle.c
index e0185e841b2a..374aa2e8203f 100644
--- a/drivers/acpi/x86/s2idle.c
+++ b/drivers/acpi/x86/s2idle.c
@@ -309,6 +309,12 @@ static void lpi_check_constraints(void)
 			continue;
 		}
 
+		if (!acpi_get_first_physical_node(adev)) {
+			acpi_handle_debug(handle, "LPI: Device is not physically present\n");
+			lpi_constraints_table[i].handle = NULL;
+			continue;
+		}
+
 		if (adev->power.state < lpi_constraints_table[i].min_dstate)
 			acpi_handle_info(handle,
 				"LPI: Constraint not met; min power state:%s current power state:%s\n",
-- 
2.35.1

